### PR TITLE
ci: remove "say" workaround and update version

### PR DESCRIPTION
**Pull request by dannysauer**
## Description

* The say command is (finally) no longer required. :) https://github.com/awslabs/git-secrets/pull/221/files
* Use a variable to represent tempdir (DRY)
* Only check out one revision of secrets tool, not all
* Switch to checking out current main/master, since upstream doesn't consistently tag releases

## Test Plan

See if this job works on this PR. :)

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by dannysauer** [comment]
Seems to have passed.  The rest of CI is really redundant, but whatever.
https://github.com/determined-ai/determined/actions/runs/5860570412/job/15888928610?pr=7628

File: .github/workflows/lint-secrets.yml [hunk 1]
```diff
@@ -18,21 +18,20 @@ jobs:
       # TODO(danh): this install+configure git-secrets is a good candidate for a
       # custom action repo -- closest to our needs I found is:
       # https://github.com/kams-mash/gh-secrets-scanner-action
+      - name: Set tempdir because github actions are dumb
+        run: echo "SECRETS_REPO=$(mktemp -dp "$PWD")" >> "$GITHUB_ENV"
```
Comment (dzhu) [inline .github/workflows/lint-secrets.yml:22]: Can we put it in `/tmp` or something? Seems better not to have extra stuff hanging out in the main repo clone.

**Review by dzhu — APPROVED** [review]
Title and commit subject could be made a bit broader, maybe.

File: .github/workflows/lint-secrets.yml (hunk 1 above)
Comment (dannysauer) [reply .github/workflows/lint-secrets.yml:22]: Turns out, we can't. I initially tried to use the runner.tmpdir value which is guaranteed clean on each run, but actions/checkout refuses to put the repo outside of the workdir. I reasoned the rm already there was equally gross as any other workaround, so left it.

Comment (dannysauer) [reply .github/workflows/lint-secrets.yml:22]: Looks like I forgot to remove that bullet from the description, too. :)

**Pull request merged**